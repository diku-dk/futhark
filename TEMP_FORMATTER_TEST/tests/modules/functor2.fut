-- Referring to a parameter-defined type in a functor return signature.
-- ==
-- input { 2 } output { 4 }

module F(P:{type t val f:t->t}): {type t = P.t val f2:t->t} = {
type t = P.t
def f2(x: t): t = P.f (P.f x)
}

module F' = F({type t = i32 def f (x: i32): i32 = x+1})

def main(x: i32): F'.t = F'.f2 x
