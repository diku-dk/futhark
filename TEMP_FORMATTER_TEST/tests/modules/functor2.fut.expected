7ecbab2c299820d32c5bd2fcde7ae001
